x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -204,6 +204,24 @@ public SemanticTextFieldMapper build(MapperBuilderContext context) {
             }
             var childContext = context.createChildContext(leafName(), ObjectMapper.Dynamic.FALSE);
             final ObjectMapper inferenceField = inferenceFieldBuilder.apply(childContext);
+
+            if (inferenceId.isConfigured() == false && searchInferenceId.isConfigured() == false) {
```

Comment:
Is it ever possible for the default inference IDs to not exist? Like, could the default inference ID be deleted? Do we need to check for this?

Reply:
AFAIU, the default inference endpoints will always exist and be immutable. We obviously can't control that here, but we work with that assumption.

@davidkyle Do I have that right?

Reply:
A 500 error might be good here for paranoia sake then.

Reply:
We can't check if the default inference endpoints actually exist here though, which would be the only scenario warranting a 500 response. This only checks if the user didn't specify a value for both the `inference_id` and `search_inference_id` mapping params.